src/main/java/com/vozniuk/bookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+package com.vozniuk.bookstore.service.impl;
+
+import com.vozniuk.bookstore.dto.cart.ShoppingCartRequestDto;
+import com.vozniuk.bookstore.dto.cart.ShoppingCartRequestDtoWithoutCartItemId;
+import com.vozniuk.bookstore.dto.cart.ShoppingCartResponseDto;
+import com.vozniuk.bookstore.exception.EntityNotFoundException;
+import com.vozniuk.bookstore.mapper.ShoppingCartMapper;
+import com.vozniuk.bookstore.model.Book;
+import com.vozniuk.bookstore.model.CartItem;
+import com.vozniuk.bookstore.model.ShoppingCart;
+import com.vozniuk.bookstore.model.User;
+import com.vozniuk.bookstore.repository.BookRepository;
+import com.vozniuk.bookstore.repository.CartItemRepository;
+import com.vozniuk.bookstore.repository.ShoppingCartRepository;
+import com.vozniuk.bookstore.repository.UserRepository;
+import com.vozniuk.bookstore.security.AuthenticationService;
+import com.vozniuk.bookstore.service.ShoppingCartService;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final UserRepository userRepository;
+    private final BookRepository bookRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final AuthenticationService authenticationService;
+
+    @Override
+    @Transactional(readOnly = true)
+    public ShoppingCartResponseDto getCart() {
+        ShoppingCart shoppingCart = shoppingCartRepository
+                .findByUserId(authenticationService.getUserId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find shopping cart by user with id: "
+                        + authenticationService.getUserId()));
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartResponseDto addToCart(ShoppingCartRequestDto requestDto) {
```

Comment:
add `Transactional` annotation for all methods that consist of more then 1 operation (in this class it should be all methods)